### PR TITLE
fix: dynamic template + context subscriptions

### LINKED ISSUE #1766: Dynamic treeshaking removing a `useWatch$` in Parent component when it shouldn't.
### Qwik Version

0.11.1

### Operating System (or Browser)

Chrome

### Node Version (if applicable)

v18

### Which component is affected?

Qwik Runtime

### Expected Behaviour

1. Create a Signal in a Parent component
2. Add a `useWatch$` for that Signal in the Parent component
3. Send the Signal down to a Child component as a prop
4. Set up a conditional in the Child component that does not utilize the Signal on the initial render (on the server)
5. Triggering something to make the Signal editable
6. Edit the Signal
7. The `useWatch$` callback should fire in the Parent component

### Actual Behaviour

1. Create a Signal in a Parent component
2. Add a `useWatch$` for that Signal in the Parent component
3. Send the Signal down to a Child component as a prop
4. Set up a conditional in the Child component that does not utilize the Signal on the initial render (on the server)
5. Triggering something to make the Signal editable
6. Edit the Signal
7. The `useWatch$` callback _does not_ fire in the Parent Component.

### Additional Information

Repro Link: https://stackblitz.com/edit/qwik-starter-2aazqq?file=src%2Froutes%2Findex.tsx,src%2Fcomponents%2FBumpCount.tsx

### PR DESCRIPTION
**Pull request by manucorporat**
# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests

# Description

Fixes #1766

### CODE REVIEW

File: packages/docs/src/components/header/header.tsx [hunk 1]
```diff
@@ -48,37 +42,24 @@ export const Header = component$(() => {
         </button>
         <ul className="md:grow md:flex md:justify-end md:p-4 menu-toolkit">
           <li>
-            <a
-              href="/docs/overview/"
-              class={{ active: pathname.startsWith('/docs') }}
-              onClick$={closeMenu}
-            >
+            <a href="/docs/overview/" class={{ active: pathname.startsWith('/docs') }}>
```
Comment (wmertens) [inline packages/docs/src/components/header/header.tsx:45]: Is the menu supposed to stay open after clicking now? Seems like that should be a separate commit.

Comment (manucorporat) [reply packages/docs/src/components/header/header.tsx:45]: it's a full page reload, no need to run JS, it's just doing nothing

File: packages/docs/src/components/docsearch/doc-search.tsx [hunk 2]
```diff
@@ -52,6 +52,8 @@ export function isEditingContent(event: KeyboardEvent): boolean {
 
 export const DocSearch = component$((props: DocSearchProps) => {
   useStyles$(styles);
+  // useContextBoundary();
```
[the remark this answers is not in the document]
Comment (manucorporat) [reply packages/docs/src/components/docsearch/doc-search.tsx:55]: ahhh! new API! but i can include it yet because we need a new release

File: packages/qwik/src/core/container/pause.ts [hunk 3]
```diff
@@ -196,25 +192,18 @@ export const _pauseFromContexts = async (
   let promises: Promise<any>[];
   while ((promises = collector.$promises$).length > 0) {
     collector.$promises$ = [];
-    await Promise.allSettled(promises);
+    await Promise.all(promises);
```
[the remark this answers is not in the document]
Comment (manucorporat) [reply packages/qwik/src/core/container/pause.ts:195]: actually the oposite, previously the error was being ignored, in fact we have this exact lines of code somewhere else in the same file and we do `Promise.all()`, `Promise.allSettled()` never rejects
